web/src/Sidebar.jsx

```diff
@@ -21,9 +21,12 @@ export default function Sidebar() {
           matches ? (
             <Fragment>
               <Separator />
-              {Object.entries(cameras).map(([camera]) => (
-                <Destination key={camera} href={`/cameras/${camera}`} text={camera} />
-              ))}
+              {Object.entries(cameras)
```

Comment:
Probably best to do this in a `useMemo` above so this logic doesn't run on every render.

Reply:
I am having a difficult time figuring out the useMemo. I think I have the memo logic figured out but it says that a memo can not be used in a conditional and only be used in the same order every time. I am assuming it is referring to the `match` condition. I looked it up and seems like I may need to split the sorted cameras part into its own function, do you have any recommendations on a clean way to do that?

Reply:
You have to call `useMemo` above the null config check in the component.

Reply:
Nevermind I think I finally got it

Reply:
Okay, let me know if what I did works. Also should I do the same in the `Cameras.jsx` as well?

Reply:
Yes.

Reply:
Cool, seems to be working well now after my latest commits 👍